### PR TITLE
#1604 Archivable Teams Edit Schema

### LINKED ISSUE #1604: Archivable Teams - Edit Schema
### Description

We need to edit the schema to add a date deleted (or date Archived) along with a userDelete (or userArchived) field

### Acceptance Criteria

Has those two fields and they are optional

### Proposed Solution

View how we do soft deletions in the db currently

### Mocks

epic #1603

### PR DESCRIPTION
**Pull request by tigerhippo**
## Changes

I added two new optional fields: dateArchived and userArchived, to the Team type in the Prisma schema. I also had to add a third new optional field, userId, because the relation for userArchived needed it. I then ran 'yarn prisma:migrate' and added a descriptive name for the migration.

## Test Cases

- yarn test:backend + frontend pass all test cases
- yarn start lets me run the FinishLine app and view it in a tab with no problems
- the migration I can see is consistent with the time it was done and the name I gave it
Screen Shot 2023-11-15 at 9 10 05 PM

## Checklist

- [X] All commits are tagged with the ticket number
- [X] No linting errors / newline at end of file warnings
- [X] All code follows repository-configured prettier formatting
- [X] No merge conflicts
- [X] All checks passing
- [X] Screenshots of UI changes (see Screenshots section)
- [X] Remove any non-applicable sections of this template
- [X] Assign the PR to yourself
- [X] No `yarn.lock` changes (unless dependencies have changed)
- [X] Request reviewers & ping on Slack
- [X] PR is linked to the ticket (fill in the closes line below)

Closes #1604

### CODE REVIEW

File: src/backend/src/prisma/schema.prisma [hunk 1]
```diff
@@ -114,18 +114,22 @@ model User {
...
+  teamName     String
+  slackId      String
+  description  String    @default("")
+  projects     Project[] @relation(name: "assignedBy")
+  members      User[]    @relation(name: "teamsAsMember")
+  leads        User[]    @relation(name: "teamsAsLead")
+  headId       Int       @unique
+  head         User      @relation(name: "teamAsHead", fields: [headId], references: [userId])
+  dateArchived DateTime?
+  userId       Int?
```
Comment (RChandler234) [inline src/backend/src/prisma/schema.prisma:131]: this should be userArchivedId

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/backend/src/prisma/schema.prisma [hunk 2]
```diff
@@ -114,20 +114,24 @@ model User {
   createdAssemblies         Assembly[]              @relation(name: "assemblyCreator")
   deletedMaterials          Material[]              @relation(name: "materialDeleter")
   createdMaterials          Material[]              @relation(name: "materialCreator")
+  Team                      Team[]
```
Comment (RChandler234) [inline src/backend/src/prisma/schema.prisma:117]: can you rename this to archivedTeams, and add a name to the relation on the team and use that with @relation(name: "name") here

File: src/backend/src/prisma/migrations/20231127023813_team_with_user_archived_fixed/migration.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/*
+  Warnings:
+
+  - You are about to drop the column `userId` on the `Team` table. All the data in the column will be lost.
+
+*/
+-- DropForeignKey
```
Comment (RChandler234) [inline src/backend/src/prisma/migrations/20231127023813_team_with_user_archived_fixed/migration.sql:7]: can you delete both of these migrations and generate a new one that has all the changes in one migration

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Comment by tigerhippo** [comment]
Awesome!

**Pull request merged**